src/orange/g4org/SolidConverter.cc

```diff
@@ -361,18 +361,23 @@ auto SolidConverter::ellipsoid(arg_type solid_base) -> result_type
 {
     auto const& solid = dynamic_cast<G4Ellipsoid const&>(solid_base);
 
-    auto rad_z = solid.GetSemiAxisMax(to_int(Axis::z));
+    auto radii = scale_.to<Real3>(solid.GetSemiAxisMax(to_int(Axis::x)),
+                                  solid.GetSemiAxisMax(to_int(Axis::y)),
+                                  solid.GetSemiAxisMax(to_int(Axis::z)));
+    auto bottom_cut = scale_.to<real_type>(solid.GetZBottomCut());
+    auto top_cut = scale_.to<real_type>(solid.GetZTopCut());
```

Comment:
The `to<Foo>` is only needed for arrays or other objects.
```suggestion
    auto bottom_cut = scale_(solid.GetZBottomCut());
    auto top_cut = scale_(solid.GetZTopCut());
```